src/deep-resource/test/Mock/CacheMock.js

```diff
@@ -1,10 +1,232 @@
 'use strict';
 
-export default class CacheMock {
+export class CacheMock {
   constructor() {
     this._container = null;
     this._driver = null;
     this._localBackend = false;
     this._microservice = null;
+    this._methodsBehavior = new Map();
+    this.enableNoResultMode();
+  }
+
+  /**
+   * @param {String} cacheKey
+   * @param {Function} callback
+   * @returns {CacheMock}
+   */
+  has(cacheKey, callback) {
+    switch (this._methodsBehavior.get('has')) {
+      case CacheMock.NO_RESULT_MODE:
+        callback(null, null);
+        break;
+
+      case CacheMock.FAILURE_MODE:
+        callback(CacheMock.ERROR, null);
+        break;
+
+      case CacheMock.DATA_MODE:
+        callback(null, CacheMock.DATA);
+        break;
+    }
```

Comment:
- create a private method and use this switch case only once (DRY - https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)